polaris-react/src/components/Text/Text.tsx

```diff
@@ -72,6 +72,7 @@ export const Text = ({
     styles.root,
     styles[variant],
     fontWeight ? styles[fontWeight] : styles[VariantFontWeightMapping[variant]],
+    (alignment !== 'inherit' || truncate) && styles.block,
```

Comment:
```suggestion
(alignment || truncate) && styles.block
```

And then you can remove `inherit` as a type from `line 21` and as a default argument on `line 60` since it's in `.root` style.